### PR TITLE
Features/7570

### PR DESCRIPTION
**Pull request by FuriousLlama**
[AB#7570](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/7570)

Created a new component for medication information
Created stores for the medication and pharmacy files. 
Updated the component to match the UX specs.

### CODE REVIEW

**Comment by healthgatewaybc** [comment]
WebClient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=webclient_pr_182)

File: Apps/Medication/src/Services/RestMedicationStatementService.cs [hunk 1]
```diff
@@ -83,7 +83,8 @@ private async Task<HNMessage<List<MedicationStatement>>> RetrieveMedicationState
         {
             string jwtString = this.httpContextAccessor.HttpContext.Request.Headers["Authorization"][0];
             string phn = await this.patientDelegate.GetPatientPHNAsync(hdid, jwtString).ConfigureAwait(true);
-            string userId = this.httpContextAccessor.HttpContext.User.Identity.Name;
+            //string userId = this.httpContextAccessor.HttpContext.User.FindFirst("hdid").Value;
+            string userId = "USER_ID";
```
Comment (TiagoGraf89) [inline Apps/Medication/src/Services/RestMedicationStatementService.cs]: did you leave this on purpose? if so lets create task to fix it in the future

File: Apps/Medication/src/Services/RestPharmacyService.cs [hunk 2]
```diff
@@ -51,7 +51,8 @@ public RestPharmacyService(IHttpContextAccessor httpAccessor, IHNClientDelegate
         public async Task<HNMessage<Pharmacy>> GetPharmacyAsync(string pharmacyId)
         {
             string jwtString = this.httpContextAccessor.HttpContext.Request.Headers["Authorization"][0];
-            string userId = this.httpContextAccessor.HttpContext.User.Identity.Name;
+            //string userId = this.httpContextAccessor.HttpContext.User.FindFirst("hdid").Value;
+            string userId = "USER_ID";
```
Comment (TiagoGraf89) [inline Apps/Medication/src/Services/RestPharmacyService.cs]: Same here

File: Apps/WebClient/src/ClientApp/test/timeline.test.ts [hunk 3]
```diff
@@ -75,6 +75,9 @@ class MockMedicationService implements IMedicationService {
       }
     });
   }
+  getMedicationInformation(drugIdentifier: string): Promise<RequestResult> {
+    throw new Error("Method not implemented.");
+  }
   getPharmacyInfo(pharmacyId: string): Promise<RequestResult> {
     return new Promise<RequestResult>((resolve, reject) => {
       if (pharmacyId === "123") {
```
Comment (TiagoGraf89) [inline Apps/WebClient/src/ClientApp/test/timeline.test.ts]: I think you need to change this to match the pharmacyIds: "pharmacyId"

File: Apps/Medication/src/Services/RestMedicationStatementService.cs (hunk 1 above)
Comment (bradhead) [reply Apps/Medication/src/Services/RestMedicationStatementService.cs]: Let's create a common function to string build a 'surname.given_names' as userId then prune it to max length in the field setter downstream.

File: Apps/Medication/src/Services/RestPharmacyService.cs (hunk 2 above)
Comment (bradhead) [reply Apps/Medication/src/Services/RestPharmacyService.cs]: same as for med statement userid

File: Apps/Medication/src/Services/RestMedicationStatementService.cs (hunk 1 above)
Comment (FuriousLlama) [reply Apps/Medication/src/Services/RestMedicationStatementService.cs]: Yep, I've created an issue for that https://qslvic.visualstudio.com/Health%20Gateway/_workitems/edit/7606

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_182)

**Comment by healthgatewaybc** [comment]
WebClient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=webclient_pr_182)

**Pull request merged**